gift-certificate-application/service/src/test/java/com/epam/esm/service/impl/OrderServiceImplTest.java

```diff
@@ -0,0 +1,177 @@
+        giftCertificate.setName(NAME);
+        giftCertificate.setDescription(DESCRIPTION);
+        giftCertificate.setPrice(PRICE);
+        giftCertificate.setDuration(DURATION);
+        giftCertificate.setTagList(tagList);
+
+        order = new Order();
+        order.setId(ID);
+        order.setUser(user);
+        order.setGiftCertificate(giftCertificate);
+        order.setPrice(giftCertificate.getPrice());
+        order.setCreateDate(LocalDateTime.now());
+
+        orderDto = new OrderDto();
+        orderDto.setUserId(USER_ID);
+        orderDto.setGiftCertificateId(GIFT_CERTIFICATE_ID);
+
+        orderList = new ArrayList<>();
+        orderList.add(order);
+
+        pagination = new CustomPagination();
+        pagination.setPage(PAGE);
+        pagination.setSize(SIZE);
+
+        orderService = new OrderServiceImpl(orderDao, userService, giftCertificateService, paginationValidator);
+    }
+
+    @Test
+    void testFindAll() {
+        List<Order> expectedResult = orderList;
+        when(orderDao.findAll(pagination, Order.class)).thenReturn(orderList);
+        when(orderDao.findEntitiesNumber(Order.class)).thenReturn(ENTITIES_NUMBER);
+        when(paginationValidator.validatePagination(pagination, ENTITIES_NUMBER)).thenReturn(pagination);
+        List<Order> actualResult = orderService.findAll(pagination);
+        assertEquals(expectedResult, actualResult);
+    }
+
+    @Test
+    void testFindById() {
+        Order expectedResult = order;
+        when(orderDao.findById(ID, Order.class)).thenReturn(Optional.ofNullable(order));
+        Order actualResult = orderService.findById(ID);
+        assertEquals(expectedResult, actualResult);
+    }
+
+    @Test
+    void testFindByIdShouldTrowOrderNotFoundException() {
+        when(orderDao.findById(ID, Order.class)).thenReturn(Optional.empty());
+        assertThrows(OrderNotFoundException.class, () -> orderService.findById(ID));
+    }
+
+    @Test
+    void testFindByUserId() {
+        List<Order> expectedResult = orderList;
+        when(userService.findById(USER_ID)).thenReturn(user);
+        when(orderDao.findByUserId(USER_ID, pagination)).thenReturn(orderList);
+        when(orderDao.findUserOrdersNumber(USER_ID)).thenReturn(ENTITIES_NUMBER);
+        when(paginationValidator.validatePagination(pagination, ENTITIES_NUMBER)).thenReturn(pagination);
+        when(orderDao.findByUserId(USER_ID, pagination)).thenReturn(orderList);
+        List<Order> actualResult = orderService.findByUserId(USER_ID, pagination);
+        assertEquals(expectedResult, actualResult);
+    }
+
+    @Test
+    void testFindByUserIdShouldTrowUserHasNoOrdersException() {
+        when(orderDao.findByUserId(USER_ID, pagination)).thenReturn(Collections.emptyList());
+        assertThrows(UserHasNoOrdersException.class, () -> orderService.findByUserId(USER_ID, pagination));
+    }
+
+    @Test
+    void testFindByAttributes() {
+        List<Order> expectedResult = orderList;
```

Comment:
I suppose you can directly use orderList and do not create an additional variable